lib/Containers/MerkleTree.cpp

```diff
@@ -723,8 +719,7 @@ template<typename Hasher, std::uint64_t const BranchingBits>
 std::vector<std::pair<std::uint64_t, std::uint64_t>>
 MerkleTree<Hasher, BranchingBits>::diff(
     MerkleTree<Hasher, BranchingBits>& other) {
-  std::shared_lock<std::shared_mutex> guard1(_dataLock);
-  std::shared_lock<std::shared_mutex> guard2(other._dataLock);
+  std::scoped_lock guard(_dataLock, other._dataLock);
```

Comment:
https://github.com/arangodb/arangodb/blob/devel/lib/Containers/MerkleTree.cpp#L530 in this line i think we have the same problem

Reply:
Good catch!